src/CFBG.cpp

```diff
@@ -655,6 +671,44 @@ bool CFBG::FillPlayersToCFBG(BattlegroundQueue* bgqueue, Battleground* bg, const
 
             return true;
         }
+
+        /* only for EvenTeams */
+        uint32 c = 0;
+        uint32 sumLevel = 0;
+        uint32 sumItemLevel = 0;
+        averagePlayersLevelQueue = 0;
+        averagePlayersItemLevelQueue = 0;
+
+        BattlegroundQueue::GroupsQueueType::const_iterator Ali_itr = bgqueue->m_QueuedGroups[bracket_id][BG_QUEUE_CFBG].begin();
+        while (Ali_itr != bgqueue->m_QueuedGroups[bracket_id][BG_QUEUE_CFBG].end() && bgqueue->m_SelectionPools[TEAM_ALLIANCE].AddGroup((*Ali_itr), aliFree))
```

Comment:
it's not clear what's happening, can you please add more comments? so other devs can understand and easily contribute too..